### PR TITLE
Journald Logger 

### PR DESCRIPTION
**Pull request by wasup-yash**
#### What type of PR is this?
/kind feature

#### What this PR does / why we need it:
This PR is to implement additional log driver support for the Journald which enables the container logs written to the systemd journal.
 
#### Which issue(s) this PR fixes: 
part of #1126 

#### Special notes for your reviewer:
None

#### Does this PR introduce a user-facing change?

```release-note
Added journald container logger support.
```

### CODE REVIEW

File: conmon-rs/server/src/container_log/journald.rs [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+            path: path.as_ref().into(),
+            file: None,
+            max_log_size,
+            bytes_written: 0,
+        })
+    }
+
+    pub async fn init(&mut self) -> Result<()> {
+        debug!("Initializing Journald logger in path {}", self.path().display());
+        self.set_file(Self::open(self.path()).await?.into());
```
Comment (haircommander) [inline conmon-rs/server/src/container_log/journald.rs:43]: hm we shouldn't be writing to a file here, it should be the system log, similar to the journald server logger

Comment (wasup-yash) [reply conmon-rs/server/src/container_log/journald.rs:43]: oh :)

**Comment by saschagrunert** [comment]
@wasup-yash may I ask you to rebase this one?

**Comment by wasup-yash** [comment]
> @wasup-yash may I ask you to rebase this one

yes surely

**Comment by saschagrunert** [comment]
@wasup-yash do you want to give this one a rebase on top of `main`?

File: conmon-rs/server/src/container_io.rs [hunk 2]
```diff
@@ -1,5 +1,5 @@
 use crate::{
-    attach::SharedContainerAttach, container_log::SharedContainerLog, streams::Streams,
+    attach::SharedContainerAttach,container_log::container_log::SharedContainerLog, streams::Streams,
```
Comment (haircommander) [inline conmon-rs/server/src/container_io.rs:2]: nit, you lost a space

File: conmon-rs/server/src/container_log/container_log.rs [hunk 3]
```diff
@@ -72,7 +88,7 @@ impl ContainerLog {
         .collect::<Result<Vec<_>>>()?;
         Ok(())
     }
-
+
```
Comment (haircommander) [inline conmon-rs/server/src/container_log/container_log.rs:91]: seems like a tab has been added

File: conmon-rs/server/src/container_log/container_log.rs [hunk 4]
```diff
@@ -108,20 +124,23 @@ impl ContainerLog {
...
             .collect::<Vec<_>>();
-
         join_all(futures)
             .await
             .into_iter()
             .collect::<Result<Vec<_>>>()?;
         Ok(())
-    }
-}
+    }
```
Comment (haircommander) [inline conmon-rs/server/src/container_log/container_log.rs:145]: space added

File: pkg/client/client.go [hunk 5]
```diff
@@ -686,6 +686,7 @@ const (
 	// type.
 	LogDriverTypeContainerRuntimeInterface LogDriverType = iota
 	LogDriverTypeJSONLogger                LogDriverType = iota
+	LogDriverTypeJournald LogDriverType = iota
```
Comment (haircommander) [inline pkg/client/client.go:689]: whitespace is off

File: pkg/client/client.go [hunk 6]
```diff
@@ -907,6 +908,9 @@ func (c *ConmonClient) initLogDrivers(req *proto.Conmon_CreateContainerRequest,
 		if logDriver.Type == LogDriverTypeJSONLogger {
 			n.SetType(proto.Conmon_LogDriver_Type_json)
 		}
+		if logDriver.Type == LogDriverTypeJournald{
```
Comment (haircommander) [inline pkg/client/client.go:911]: space between `d{`

**Comment by grassdionera** [comment]
would love to see this merged. then cri-o/cri-o can support journald logging when they switch to common-rs

**Comment by saschagrunert** [comment]
@haircommander @kwilczynski I rebased this one and added some fixes. I guess we can follow-up with integration tests in a dedicated PR, but that's now ready to get merged from my point of view. PTAL.

**Comment by haircommander** [comment]
I worry about merging before integration tests TBH 
/approve
/lgtm
/hold

You can lift the hold if you'd like, but we should get the tests ASAP to make sure the structure is correct IMO

**Comment by rphillips** [comment]
I'm good with merging. Looks like we default to the CRI logger anyway.
+1 to integration tests as a followup.

**Comment by saschagrunert** [comment]
@haircommander I added an integration test to ensure the correct functionality.

**Comment by kwilczynski** [comment]
/hold

So I can make a review.

**Comment by rphillips** [comment]
@haircommander memory usage ok?

**Comment by saschagrunert** [comment]
@haircommander PTAL

**Pull request merged**